contracts/interfaces/ISafe.sol

```diff
@@ -0,0 +1,146 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity >=0.7.0 <0.9.0;
+
+import {Enum} from "./IEnum.sol";
+
+/**
+ * @title ISafe - A multisignature wallet interface with support for confirmations using signed messages based on EIP-712.
+ * @author Shebin John - @remedcu
+ */
+interface ISafe {
+    /**
+     * @notice Sets an initial storage of the Safe contract.
+     * @dev This method can only be called once.
+     *      If a proxy was created without setting up, anyone can call setup and claim the proxy.
+     * @param _owners List of Safe owners.
+     * @param _threshold Number of required confirmations for a Safe transaction.
+     * @param to Contract address for optional delegate call.
+     * @param data Data payload for optional delegate call.
+     * @param fallbackHandler Handler for fallback calls to this contract
+     * @param paymentToken Token that should be used for the payment (0 is ETH)
+     * @param payment Value that should be paid
+     * @param paymentReceiver Address that should receive the payment (or 0 if tx.origin)
+     */
+    function setup(
```

Comment:
should the safe contract inherit the docs from here (@inheritdoc), so we don't have to maintain two pieces of the documentation

Reply:
As `ISafe` is not inherited by `Safe`, using `solc` to create the documentation with this tag won't work. Tried using the command `solc --userdoc ./contracts/Safe.sol`

Reply:
What if we inherit it? Other solid codebases like [Seaport](https://github.com/ProjectOpenSea/seaport) do that

Reply:
It won't compile because of the other inherited functions specified in ISafe: https://github.com/safe-global/safe-contracts/pull/722/files#diff-850f06cc1a3d3ce68df28ca73abce16a58e42ba86c15ce3667a3ecd8d7155441R138 (If there is any other workaround, please do let me know).

Reply:
Made changes based on the first suggestion. For interfaces, all methods are `virtual` implicitly.

Done 👍🏾